src/sync/arc.rs

```diff
@@ -5,36 +5,72 @@ use std::{mem, ops};
 
 /// Mock implementation of `std::sync::Arc`.
 #[derive(Debug)]
-pub struct Arc<T> {
-    inner: std::sync::Arc<Inner<T>>,
-}
-
-#[derive(Debug)]
-#[repr(C)]
-struct Inner<T> {
-    // This must be the first field to make into_raw / from_raw work
-    value: T,
-
-    obj: rt::Arc,
+pub struct Arc<T: ?Sized> {
+    obj: std::sync::Arc<rt::Arc>,
+    inner: std::sync::Arc<T>,
```

Comment:
nit, take it or leave it: should we maybe consider renaming this field from `inner` to `value` or something? since it now stores only the actual `Arc`ed value, it might be worth making that clearer in the `Debug` output? I note that the field holding the Arc's value used to be `inner.value`...